sphinx_gallery/gen_gallery.py

```diff
@@ -687,6 +692,150 @@ def write_computation_times(gallery_conf, target_dir, costs):
+                            used_api_entries[entry].append(example_name)
+
+        title = 'Unused API Entries'
+        fid.write(title + '\n' + '=' * len(title) + '\n\n')
+        for entry in sorted(unused_api_entries):
+            fid.write(f'- :{get_entry_type(entry)}:`{entry}`\n')
+        fid.write('\n\n')
+
+        unused_dot_fname = os.path.join(target_dir, 'sg_api_unused.dot')
+        if has_graphviz and unused_api_entries:
+            fid.write('.. graphviz:: ./sg_api_unused.dot\n'
+                      '    :alt: API used entries graph\n'
+                      '    :layout: neato\n\n')
+
+        used_count = len(used_api_entries)
+        used_percentage = used_count / total_count
+        fid.write('\nAPI entries used: '
+                  f'{round(used_percentage * 100, 2)}% '
+                  f'({used_count}/{total_count})\n\n')
+
+        title = 'Used API Entries'
+        fid.write(title + '\n' + '=' * len(title) + '\n\n')
+        for entry in sorted(used_api_entries):
+            fid.write(f'- :{get_entry_type(entry)}:`{entry}`\n\n')
+            for ref in used_api_entries[entry]:
+                fid.write(f'  - :ref:`{ref}`\n')
+            fid.write('\n\n')
+
+        used_dot_fname = os.path.join(target_dir, 'sg_api_used.dot')
+        if has_graphviz and used_api_entries:
+            fid.write('.. graphviz:: ./sg_api_used.dot\n'
+                      '    :alt: API usage graph\n'
+                      '    :layout: neato\n\n')
+
+    # design graph
+    if has_graphviz and unused_api_entries:
+        dg = graphviz.Digraph('api_usage', filename=unused_dot_fname,
+
+                              node_attr={'color': 'lightblue2',
+                                         'style': 'filled'})
+
+        unused_api_connections = set()
+        unused_api_struct = [entry.split('.') for entry in unused_api_entries]
+        n_levels = max([len(struct) for struct in unused_api_struct])
+        for level in range(n_levels):
+            for struct in unused_api_struct:
+                if len(struct) <= level + 1:
+                    continue
+                if (struct[level], struct[level + 1]) in \
+                        unused_api_connections:
+                    continue
+                unused_api_connections.add((struct[level], struct[level + 1]))
+                dg.edge(struct[level], struct[level + 1])
+
+        dg.attr(overlap='scale')
+        dg.attr(fontsize='32')
+        dg.save(unused_dot_fname)
+
+    if has_graphviz and used_api_entries:
+        dg = graphviz.Digraph('api_usage', filename=used_dot_fname,
+                              node_attr={'color': 'lightblue2',
+                                         'style': 'filled'})
+
+        for entry, refs in used_api_entries.items():
+            for ref in refs:
+                dg.edge(entry, ref[replace_count:])
+
+        dg.attr(overlap='scale')
+        dg.attr(fontsize='32')
+        dg.save(used_dot_fname)
```

Comment:
this could be a little more DRY, does much of the same stuff as the `unused_api_entries` conditional